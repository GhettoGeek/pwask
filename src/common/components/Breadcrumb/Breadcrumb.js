import React from 'react'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router-dom'
import { Breadcrumbs } from '@material-ui/core'
import { Home as HomeIcon } from '@material-ui/icons'
import { styled } from '../../utils/style'
import Link from '../Link'
import Typography from '../Typography'

const StyledHomeIcon = styled(HomeIcon)((theme) => ({
  marginRight: theme.spacing(1),
  width: 20,
  height: 20,
}))

const BreadCrumb = ({ location }) => {
  const crumbs = location.pathname.split('/').filter(Boolean)
  const crumbsLength = crumbs.length

  return (
    <Breadcrumbs aria-label="Breadcrumb">
      <Link color="inherit" href="/">
        <StyledHomeIcon />
      </Link>
      {
        crumbs.map((item, index) => {
          if (crumbsLength === index + 1) {
            return <Typography color="primary" key={index}>{item}</Typography>
          }

          return (
            <Link color="prmary" href={`/${item}`} key={index}>
              {item}
            </Link>
          )
        })
      }
    </Breadcrumbs>
  )
}

BreadCrumb.propTypes = {
  location: PropTypes.shape().isRequired,
}

export default withRouter(BreadCrumb)
