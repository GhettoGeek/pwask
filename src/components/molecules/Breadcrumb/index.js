import React from 'react'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router-dom'
import Link from '@material-ui/core/Link'
import Typography from '@material-ui/core/Typography'
import Breadcrumbs from '@material-ui/lab/Breadcrumbs'
import HomeIcon from '@material-ui/icons/Home'
import { styled } from '../../../utils/style'

const StyledHomeIcon = styled(HomeIcon)(theme => ({
  marginRight: theme.spacing.unit / 2,
  width: 20,
  height: 20,
}))

const BreadCrumb = ({ location }) => {
  const crumbs = location.pathname.split('/').filter(Boolean)
  const crumbsLength = crumbs.length

  return (
    <Breadcrumbs aria-label="Breadcrumb">
      <Link color="inherit" href="/">
        <StyledHomeIcon />
      </Link>
      {
        crumbs.map((item, index) => {
          if (crumbsLength === index + 1) {
            return <Typography color="primary" key={index}>{item}</Typography>
          }

          return (
            <Link color="prmary" href={`/${item}`} key={index}>
              {item}
            </Link>
          )
        })
      }
    </Breadcrumbs>
  )
}

BreadCrumb.propTypes = {
  location: PropTypes.shape().isRequired,
}

export default withRouter(BreadCrumb)
