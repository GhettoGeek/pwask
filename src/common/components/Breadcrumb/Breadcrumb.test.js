import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { shallow } from 'enzyme'
import Breadcrumb from './Breadcrumb'

describe('Breadcrumb', () => {
  it('renders correctly the component', () => {
    const wrapper = shallow(
      <MemoryRouter initialEntries={['/']} initialIndex={0}>
        <Breadcrumb />
      </MemoryRouter>,
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
})
