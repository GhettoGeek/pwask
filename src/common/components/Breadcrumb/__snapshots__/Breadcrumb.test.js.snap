// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb renders correctly the component 1`] = `"<nav class=\\"MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary\\" aria-label=\\"Breadcrumb\\"><ol class=\\"MuiBreadcrumbs-ol\\"><li class=\\"MuiBreadcrumbs-li\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit\\" href=\\"\\"><svg class=\\"MuiSvgIcon-root StyledComponent-root-41\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z\\"></path></svg></a></li></ol></nav>"`;
